Don't clear window style when minimizing it (MS Windows)

puchik opened the pull request:
- When the window is minimized, do not clear the window styling to avoid unexpected defaults when restoring it. (Exclude WINDOW_MODE_MINIMIZED from being a valid state for clear)
- Also, do not make borders visible on full screen. This fixes a state where the game may be full screen but the border still appears when restoring a window (there is no "bordered full screen" mode anyway)

Fixes #79641 

<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

puchik committed:
Don't clear window style when minimizing (MS Windows)

- When the window is minimized, do not clear the window styling to avoid unexpected defaults when restoring it. (Exclude WINDOW_MODE_MINIMIZED from being a valid state for clear)
- Also, do not make borders visible on full screen. This fixes a state where the game
may be full screen but the border still appears when restoring a window (there is no "bordered full screen" mode anyway)

Ybaedbaenn commented:
Thank you for fixing!

Null-Ping commented:
Thanks!

Get-Newsletters-Int commented:
Amazing work!

Get-Newsletters-Int commented:
@bruvzg

Calinou reviewed (commented):
Tested locally (rebased on top of `master` https://github.com/godotengine/godot/commit/84dcdde9088d512efebb0033fb881d8f7fd9c107), it mostly works as expected on Windows 11 23H2.

Window mode is now restored correctly using the MRP from https://github.com/godotengine/godot/issues/79641, but the issue is that the window size is incorrect when going fullscreen again:

![Screenshot 2024-09-18 144952](https://github.com/user-attachments/assets/a8dba20a-b0d2-4fc7-b7bb-c7ed5e9bb400)

If I alt-tab out and back in, the issue is revealed:

![Screenshot 2024-09-18 144956](https://github.com/user-attachments/assets/9d0ef218-0b7c-4bae-a48b-88feb8588897)

It's interesting that this Vista-style window border is even visible on Windows 11 in the first place...

puchik commented:
So I pulled and rebased when you made that comment, and it was indeed broken, and I planned to take another look later. Now I pulled and rebased, and it's working again😄. In fact, I undid all of my changes, and it was still working.

I can't reproduce this on master or the latest 4.5 release, so it may have been fixed as a side effect of something else or a Windows update. Running Windows 10 22H2 here. This might not be needed at all anymore if someone else can confirm 🙂

Calinou commented:
> I can't reproduce this on master or the latest 4.4 release, so it may have been fixed as a side effect of something else or a Windows update. Running Windows 10 22H2 here. This might not be needed at all anymore if someone else can confirm 🙂

This has been reported on 4.4... but on Linux: https://github.com/godotengine/godot/issues/79641#issuecomment-2754074459
